src/metabase/lib/equality.cljc

```diff
@@ -360,14 +360,12 @@
   Returns the matching ref, or nil if no plausible matches are found."
   [column :- ::lib.schema.metadata/column
    refs   :- [:sequential ::lib.schema.ref/ref]]
-  (let [ref-tails (group-by ref-id-or-name refs)
-        matches   (or (some->> column :lib/source-uuid (get ref-tails) not-empty)
-                      (not-empty (get ref-tails (:id column)))
-                      ;; columns from the previous stage have unique `:lib/desired-column-alias` but not `:name`.
-                      ;; we cannot fallback to `:name` when `:lib/desired-column-alias` is set
-                      (get ref-tails (or (:lib/desired-column-alias column)
-                                         (:name column)))
-                      [])]
+  (let [ref-tails       (group-by ref-id-or-name refs)
+        column-ref-tail (-> column lib.ref/ref ref-id-or-name)
```

Comment:
This is needed because now the tail can be either `:lib/desired-column-alias` OR `:name`, and we cannot match for them both